### DISCUSSION TITLE
performance tracing for serversless functions

### DISCUSSION THREAD

**Discussion by MaxBittker** [vercel/community#694] [Help] [answered]
is there a good to get a flame graph for my function execution? I'm trying to speed them up and it's hard to do it without visibility

**Comment by swarnava** [comment, marked as answer]
The Vercel platform, by default, does not store logs that derive from static assets and [Serverless Functions](https://vercel.com/docs/serverless-functions/introduction). To store logs in Vercel, we recommend taking advantage of [Log Drains](https://vercel.com/docs/logs) through 3rd-party solutions.

https://vercel.com/support/articles/how-do-i-store-logs-on-vercel

**Comment by MaxBittker** [comment]
sorry @swarnava , i'm asking not about logs, but about performance traces

**Reply by swarnava** [reply]
Most of logdrains provider should also give you details about Performance tracing. Here is an article for Datadog: https://www.datadoghq.com/product/apm/
